File: LeetCode/Math/Python/Leetcode479-最大回文数乘积.py
class Solution:
    def largestPalindrome(self, n: int) -> int:
        if n == 1: return 9
        base = (10 ** n) - 1
        for i in range(base,-1,-1):
            p,x = i,i
            while x != 0:
                p = p * 10 + x % 10
                x //= 10
            x = base
            while x >= p // x:
                if p % x == 0:
                    return p % 1337
                x -= 1 

File: LeetCode/Math/Golang/Leetcode479-最大回文数乘积.go
package sbgo

import "math"

func largestPalindrome(n int) int {
	if n == 1 {
		return 9
	}
	base := int(math.Pow10(n) - 1)

	for x := base; ; x-- {
		p := x
		for y := x; y != 0; y /= 10 {
			p = p*10 + y%10
		}

		for y := base; y >= p/y; y-- {
			if p%y == 0 {
				return p % 1337
			}
		}
	}
}
